transports/webrtc-websys/src/transport.rs

```diff
@@ -99,3 +107,34 @@ impl libp2p_core::Transport for Transport {
         None
     }
 }
+
+/// Checks if local Firefox.
+///
+/// See: https://bugzilla.mozilla.org/show_bug.cgi?id=1659672 for more details
+fn maybe_local_firefox() -> bool {
+    let window = &web_sys::window().expect("window should be available");
+    let ua = match window.navigator().user_agent() {
+        Ok(agent) => agent.to_lowercase(),
+        Err(_) => return false,
+    };
+
+    let hostname = match window
+        .document()
+        .expect("should be valid document")
+        .location()
+    {
+        Some(location) => match location.hostname() {
+            Ok(hostname) => hostname,
+            Err(_) => return false,
+        },
+        None => return false,
+    };
+
+    // check if web_sys::Navigator::user_agent() matches any of the following:
+    // - firefox
+    // - seamonkey
+    // - iceape
+    // AND hostname is either localhost or  "127.0.0.1"
+    (ua.contains("firefox") || ua.contains("seamonkey") || ua.contains("iceape"))
+        && (hostname == "localhost" || hostname == "127.0.0.1")
```

Comment:
Would the same be relevant for `::1`?

Reply:
I suppose, but I don't know a lot of people who use `http://[::1]:8080`
Think it's worthwhile to cover?

Reply:
I wonder if hostname IPv6 would even come in as `hostname == "::1"` or `hostname == "[::1]"`

Reply:
I will add this in:
```
window.location.hostname
'[::1]'
```